src/components/NcFieldset/NcFieldset.vue

```diff
@@ -0,0 +1,174 @@
+<!--
+  - SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<script setup lang="ts">
+import type { Slot } from 'vue'
+
+import { createElementId } from '../../utils/createElementId.ts'
+
+const {
+	label = undefined,
+	description = undefined,
+	hideLabel = false,
+	hideDescription = false,
+	noGap = false,
+} = defineProps<{
+	/**
+	 * Group label (the fieldset's legend). #label slot can be used for custom label content
+	 */
+	label?: string
+	/**
+	 * Optional fieldset description. #description slot can be used for custom description content
+	 */
+	description?: string
+	/**
+	 * Hide the label visually but keep it accessible for screen readers
+	 */
+	hideLabel?: boolean
+	/**
+	 * Hide the description visually but keep it accessible for screen readers
+	 */
+	hideDescription?: boolean
+	/**
+	 * Disable default fieldset content gap between content elements
+	 */
+	noGap?: boolean
+}>()
+
+const slots = defineSlots<{
+	/**
+	 * Fieldset content
+	 */
+	default?: Slot
+	/**
+	 * Custom label content
+	 */
+	label?: Slot
+	/**
+	 * Custom description content
+	 */
+	description?: Slot
+}>()
+
+const id = `nc-fieldset-${createElementId()}`
+const labelId = `${id}-label`
+const descriptionId = `${id}-description`
+
+const hasDescription = () => !!description || !!slots.description
+const getDescriptionId = () => hasDescription() ? descriptionId : undefined
+</script>
+
+<template>
+	<fieldset
+		:class="[$style.fieldset, { [$style.fieldset_noGap]: noGap }]"
+		:aria-describedby="getDescriptionId()">
+		<legend :id="labelId" :class="[$style.fieldset__legend, { 'hidden-visually': hideLabel }]">
+			<slot name="label">
+				{{ label || '⚠️ Missing label' }}
```

Comment:
Do we want to always have a label? I can imagine a use-case like having a Fieldset inside a Modal which already has a title. I feel something might eventually come up where you would want to just not have a label, and having it show "Missing label" would be uncomfortable.

If we want to always have a standard label as a design decision though, I could understand it.

Reply:
> Do we want to always have a label? 

If its a fieldset we must. Its obligatory for wcag. If its only visually then its the form box not a real group.